cli/tests/wpt_testharnessconsolereporter.js

```diff
@@ -0,0 +1,119 @@
+const noColor = globalThis.Deno?.noColor ?? true;
+const enabled = !noColor;
+
+function code(open, close) {
+  return {
+    open: `\x1b[${open.join(";")}m`,
+    close: `\x1b[${close}m`,
+    regexp: new RegExp(`\\x1b\\[${close}m`, "g"),
+  };
+}
+
+function run(str, code) {
+  return enabled
+    ? `${code.open}${str.replace(code.regexp, code.open)}${code.close}`
+    : str;
+}
+
+function red(str) {
+  return run(str, code([31], 39));
+}
+
+export function green(str) {
+  return run(str, code([32], 39));
+}
+
+export function yellow(str) {
+  return run(str, code([33], 39));
+}
+
+const testResults = [];
+const testsExpectFail = JSON.parse(Deno.args[0]);
+
+window.add_result_callback((test) => {
```

Comment:
```suggestion
window.add_result_callback(({ message, name, stack, status }) => {
```

And then you can just use them destructured below...